### PR TITLE
[directxtk, directxtk12, directxtex, directxmesh, uvatlas] Update ports for May 2022 release

### PR DESCRIPTION
**Pull request by walbourn**
DirectX Tool Kit for DX11 / DX12, DirectXTex, DirectXMesh, and UVAtlas all have new May 2022 releases on GitHub. In addition to a few bug fixes and minor features, this revision adds support for building with the MinGW toolset via CMake.

All ports now support building with **x86-mingw-static** and **x64-mingw-static** community triplets when using MinGW GNU 11. The directxtk, directxtk12, and directxtex ports also need the HLSL compiler in the path along with the appropriate compiler toolset. These ports all take advantage of the recent changes to directx-headers including conformance fixes for MinGW (#24490).

> Technically **x86-mingw-dynamic** and **x64-mingw-dynamic** community triplets also work, but they just end up switching back to static library linkage so they have the same result.

The **directxtk12** port now exposes an _xaudio2-9_ and _xaudio2redist_ feature because MinGW requires xaudio2redist to build (if xaudio2-9 is specified, the upstream CMake just turns it off if using MinGW). Therefore, to get *DirectX Tool Kit for Audio* support with either the **directxtk** or **directxtk12** ports when using MinGW, you need to use the xaudio2redist feature.

### CODE REVIEW

File: ports/directxmesh/portfile.cmake [hunk 1]
```diff
@@ -30,14 +30,14 @@ vcpkg_cmake_configure(
 )
 
 vcpkg_cmake_install()
-vcpkg_cmake_config_fixup(CONFIG_PATH cmake)
+vcpkg_cmake_config_fixup(CONFIG_PATH share/directxmesh/cmake)
```
Comment (walbourn) [inline ports/directxmesh/portfile.cmake:33]: The upstream CMakeLists now make use of ``include(GNUInstallDirs)`` so these locations changed.

**Comment by JonLiu1993** [comment]
All features are tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

**Review by JonLiu1993 — APPROVED** [review]

**Comment by walbourn** [comment]
Validated building all supported features and triplets locally:

```
vcpkg install directxmesh:x86-windows
vcpkg install directxmesh[dx12]:x86-windows --recurse
vcpkg install directxmesh:x86-windows-static
vcpkg install directxmesh:x86-windows-static-md
vcpkg install directxmesh:x64-windows
vcpkg install directxmesh[dx12]:x64-windows --recurse
vcpkg install directxmesh:x64-windows-static
vcpkg install directxmesh:x64-windows-static-md
vcpkg install directxmesh:arm-windows
vcpkg install directxmesh:arm64-windows
vcpkg install directxmesh:arm64-windows-static
vcpkg install directxmesh:arm64-windows-static-md
vcpkg install directxmesh:x86-uwp
vcpkg install directxmesh:x64-uwp
vcpkg install directxmesh:arm-uwp
vcpkg install directxmesh:arm64-uwp
vcpkg install directxmesh:x64-mingw-static
vcpkg install directxmesh[dx12]:x64-mingw-static --recurse
vcpkg install directxmesh:x86-mingw-static
vcpkg install directxmesh[dx12]:x86-mingw-static --recurse
./vcpkg install directxmesh:x64-linux
./vcpkg install directxmesh[dx12]:x64-linux --recurse
./vcpkg install directxmesh:arm64-linux
./vcpkg install directxmesh[dx12]:arm64-linux --recurse

vcpkg install directxtex:x86-windows
vcpkg install directxtex[dx12]:x86-windows --recurse
vcpkg install directxtex[openexr]:x86-windows --recurse
vcpkg install directxtex:x86-windows-static
vcpkg install directxtex:x86-windows-static-md
vcpkg install directxtex:x64-windows
vcpkg install directxtex[dx12]:x64-windows --recurse
vcpkg install directxtex[openexr]:x64-windows --recurse
vcpkg install directxtex:x64-windows-static
vcpkg install directxtex:x64-windows-static-md
vcpkg install directxtex:arm-windows
vcpkg install directxtex:arm64-windows
vcpkg install directxtex:arm64-windows-static
vcpkg install directxtex:arm64-windows-static-md
vcpkg install directxtex:x86-uwp
vcpkg install directxtex:x64-uwp
vcpkg install directxtex:arm-uwp
vcpkg install directxtex:arm64-uwp
vcpkg install directxtex:x64-mingw-static
vcpkg install directxtex[dx12]:x64-mingw-static --recurse
vcpkg install directxtex:x86-mingw-static
vcpkg install directxtex[dx12]:x86-mingw-static --recurse
./vcpkg install directxtex:x64-linux
./vcpkg install directxtex[dx12]:x64-linux --recurse
./vcpkg install directxtex[openexr]:x64-linux --recurse
./vcpkg install directxtex:arm64-linux
./vcpkg install directxtex[dx12]:arm64-linux --recurse
./vcpkg install directxtex[openexr]:arm64-linux --recurse

vcpkg install directxtk:x86-windows
vcpkg install directxtk[xaudio2-8]:x86-windows --recurse
vcpkg install directxtk[xaudio2-9]:x86-windows --recurse
vcpkg install directxtk[xaudio2redist]:x86-windows --recurse
vcpkg install directxtk[xaudio2redist]:x86-windows-static-md --recurse
vcpkg install directxtk:x86-windows-static
vcpkg install directxtk:x86-windows-static-md
vcpkg install directxtk:x64-windows
vcpkg install directxtk[xaudio2-8]:x64-windows --recurse
vcpkg install directxtk[xaudio2-9]:x64-windows --recurse
vcpkg install directxtk[xaudio2redist]:x64-windows --recurse
vcpkg install directxtk[xaudio2redist]:x64-windows-static-md --recurse
vcpkg install directxtk:x64-windows-static
vcpkg install directxtk:x64-windows-static-md
vcpkg install directxtk:arm-windows
vcpkg install directxtk:arm64-windows
vcpkg install directxtk:arm64-windows-static
vcpkg install directxtk:arm64-windows-static-md
vcpkg install directxtk:x86-uwp
vcpkg install directxtk:x64-uwp
vcpkg install directxtk:arm-uwp
vcpkg install directxtk:arm64-uwp
vcpkg install directxtk:x64-mingw-static
vcpkg install directxtk[xaudio2redist]:x64-mingw-static
vcpkg install directxtk:x86-mingw-static
vcpkg install directxtk[xaudio2redist]:x86-mingw-static

vcpkg install directxtk12[core]:x86-windows
vcpkg install directxtk12[xaudio2-9]:x86-windows --recurse
vcpkg install directxtk12[xaudio2redist]:x86-windows --recurse
vcpkg install directxtk12[xaudio2redist]:x86-windows-static-md --recurse
vcpkg install directxtk12:x86-windows-static
vcpkg install directxtk12:x86-windows-static-md
vcpkg install directxtk12[core]:x64-windows
vcpkg install directxtk12[xaudio2-9]:x64-windows --recurse
vcpkg install directxtk12[xaudio2redist]:x64-windows --recurse
vcpkg install directxtk12[xaudio2redist]:x64-windows-static-md --recurse
vcpkg install directxtk12:x64-windows-static
vcpkg install directxtk12:x64-windows-static-md
vcpkg install directxtk12:arm-windows
vcpkg install directxtk12:arm64-windows
vcpkg install directxtk12:arm64-windows-static
vcpkg install directxtk12:arm64-windows-static-md
vcpkg install directxtk12:x86-uwp
vcpkg install directxtk12:x64-uwp
vcpkg install directxtk12:arm-uwp
vcpkg install directxtk12:arm64-uwp
vcpkg install directxtk12:x64-mingw-static
vcpkg install directxtk12[xaudio2redist]:x64-mingw-static
vcpkg install directxtk12:x86-mingw-static
vcpkg install directxtk12[xaudio2redist]:x86-mingw-static

vcpkg install uvatlas:x86-windows
vcpkg install uvatlas[eigen]:x86-windows --recurse
vcpkg install uvatlas:x86-windows-static
vcpkg install uvatlas:x86-windows-static-md
vcpkg install uvatlas:x64-windows
vcpkg install uvatlas[eigen]:x64-windows --recurse
vcpkg install uvatlas:x64-windows-static
vcpkg install uvatlas:x64-windows-static-md
vcpkg install uvatlas:arm-windows
vcpkg install uvatlas:arm64-windows
vcpkg install uvatlas:arm64-windows-static
vcpkg install uvatlas:arm64-windows-static-md
vcpkg install uvatlas:x86-uwp
vcpkg install uvatlas:x64-uwp
vcpkg install uvatlas:arm-uwp
vcpkg install uvatlas:arm64-uwp
vcpkg install uvatlas:x64-mingw-static
vcpkg install uvatlas:x86-mingw-static
./vcpkg install uvatlas:x64-linux
./vcpkg install uvatlas[eigen]:x64-linux --recurse
./vcpkg install uvatlas:arm64-linux
./vcpkg install uvatlas[eigen]:arm64-linux --recurse
```

**Comment by walbourn** [comment]
Building directxtk, direcxtk12, or directxtex with MinG-W32 requires:

```
set PATH=%PATH%;c:\mingw32\bin;"c:\Program Files (x86)\Windows Kits\10\bin\10.0.22000.0\x86"
```


Building directxtk, direcxtk12, or directxtex with MinG-W64 requires:

```
set PATH=%PATH%;c:\mingw64\bin;"c:\Program Files (x86)\Windows Kits\10\bin\10.0.22000.0\x64"
```

**Comment by BillyONeal** [comment]
Thanks for the updates!

**Pull request merged**